### PR TITLE
feat: allow choose to display from tensor or from uri in document

### PR DESCRIPTION
**Pull request by samsja**
# context

This pr refactor the code so that we can chose if we want to display from uri or from tensor ( issue https://github.com/jina-ai/docarray/issues/517)

Now you can do

```python
from docarray import Document

d = Document(uri=os.path.join(cur_dir, 'toydata/test.png'))
d.display()
d.display(from_='uri')

d.load_uri_to_image_tensor()
d.uri = None

d.display()
d.display(from_='tensor')
```

### CODE REVIEW

File: docarray/document/mixins/plot.py [hunk 1]
```diff
@@ -70,38 +70,68 @@ def _plot_recursion(self, tree=None):
                     d._plot_recursion(_match_tree)
         return tree
 
-    def display(self):
-        """Plot image data from :attr:`.tensor` or :attr:`.uri`."""
-        from IPython.display import Image, display
+    def display(self, from_: Optional[str] = None):
```
Comment (alaeddine-13) [inline docarray/document/mixins/plot.py:73]: why is it with underscore ?

Comment (samsja) [reply docarray/document/mixins/plot.py:73]: `from` is a restricted keyword in python

Comment (alaeddine-13) [reply docarray/document/mixins/plot.py:73]: Okay I'm not sure what is the best practice here, @JoanFM  any idea ?

File: docarray/document/mixins/plot.py [hunk 2]
```diff
@@ -70,38 +70,68 @@ def _plot_recursion(self, tree=None):
...
+                self.summary()
+
+        if from_ == 'uri':
+            self.display_uri()
+        elif from_ == 'tensor':
+            self.display_tensor()
         else:
             self.summary()
 
+    def display_tensor(self):
```
Comment (alaeddine-13) [inline docarray/document/mixins/plot.py:95]:
```suggestion
    def _display_tensor(self):
```

File: docarray/document/mixins/plot.py [hunk 3]
```diff
@@ -70,38 +70,68 @@ def _plot_recursion(self, tree=None):
...
-
-                p = PIL.Image.fromarray(self.tensor)
-                if p.mode != 'RGB':
-                    raise
-                display(p)
-            except:
-                import matplotlib.pyplot as plt
-
-                plt.matshow(self.tensor)
+                self.summary()
```
[the remark this answers is not in the document]
Comment (samsja) [reply docarray/document/mixins/plot.py:86]: No I want it to be public.

My point is that I don't like the `display(from_='tensor')` it is just weird way of calling what should be `dispaly_tensor`. Making the method public allow to do both

[the remark this answers is not in the document]
Comment (alaeddine-13) [reply docarray/document/mixins/plot.py:86]: alright as you like

**Review by alaeddine-13 — APPROVED** [review]

**Comment by alexcg1** [comment]
How come it's `from_` and not `from`?

**Comment by JohannesMessner** [comment]
`from` is a Python keyword and cannot be used

**Pull request merged**